File: sic-api/Entities/Db/DbProvince.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using sic_api.Entities;

namespace sic_api.Entities.Db;

[Index(nameof(ProvinceCode), IsUnique = true)]
[Table("db_province")]
public class DbProvince : BaseEntity
{
    
    [Required]
    [ForeignKey(nameof(Country))]
    [Column("country_id")]
    public Guid CountryId { get; set; }
    public DbCountry Country { get; set; } = default!;
    
    [Required]
    [MaxLength(10)]
    [Column("province_code")]
    public string ProvinceCode { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("province_name_en")]
    public string ProvinceNameEn { get; set; } = default!;

    [Required]
    [MaxLength(255)]
    [Column("province_name_local")]
    public string ProvinceNameLocal { get; set; } = default!;

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    public ICollection<DbDistrict>? Districts { get; set; }
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbProvince.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "db_province",
       indexes = {
           @Index(name = "idx_province_code", columnList = "province_code", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbProvince extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "country_id", nullable = false)
    private DbCountry country;

    @Column(name = "province_code", nullable = false, length = 10)
    private String provinceCode;

    @Column(name = "province_name_en", nullable = false, length = 255)
    private String provinceNameEn;

    @Column(name = "province_name_local", nullable = false, length = 255)
    private String provinceNameLocal;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @OneToMany(mappedBy = "province") // สมมติว่า DbDistrict มีฟิลด์ province
    private List<DbDistrict> districts = new ArrayList<>();
}